mmv1/third_party/terraform/services/storage/resource_storage_bucket.go.erb

```diff
@@ -1401,11 +1414,14 @@ func expandStorageBucketLifecycleRuleCondition(v interface{}) (*storage.BucketLi
 
 	condition := conditions[0].(map[string]interface{})
 	transformed := &storage.BucketLifecycleRuleCondition{}
-
-	if v, ok := condition["age"]; ok {
-		age := int64(v.(int))
-		transformed.Age = &age
-		transformed.ForceSendFields = append(transformed.ForceSendFields, "Age")
+	// Setting high precedence of no_age over age when both used together.
+	// Only sets age value to transformed when no_age has false value
+	if v, ok := condition["no_age"]; ok && !(v.(bool)) {
```

Comment:
I'm wondering about the case where `no-age` is not set. It appears to be covered properly in your tests, but I was expecting to see the `age` being set even if `ok` is false.

Reply:
`no_age` attribute behavior is similar to `age` attribute behavior. It is always getting false value because nil value for boolean is false and this attribute is also inside of `TypeSet`. Removal of `ok` from if condition won't cause issue now but if in future, this issue of getting value every time even when not set, gets properly handle by terraform team then provider starts breaking because of this. We are using v.bool value as condition and suddenly we are not getting no_age from `condition` which will cause more problems.

Reply:
I'm having a little bit of trouble following on this point. Do you happen to have a link to the Terraform bug you're referencing?

It seems to me you're talking about the treatment of zero-values (ie. an unset `no_age` evaluates to false). Is that correct? If so, then we rely on that behavior heavily in the provider, and I wouldn't worry about that changing (it might in the future, if we switch to the plugin framework, but it would be a large change across the entire provider). I believe the case where `no_age` is not set will behave properly either way here, but accounting for the `!ok` case would help make sure we are covering all branches, and is hypothetically more resilient to things like fields moving.

Reply:
Okay! In that case i will modify condition like this `!ok || !(v.(bool))` which check that `no_age` is not found or if it found then it has false value.